lib/crypto.js

```diff
@@ -537,21 +537,16 @@ ECDH.prototype.getPublicKey = function getPublicKey(encoding, format) {
 };
 
 
-const pbkdf2DeprecationWarning =
-    internalUtil.deprecate(() => {}, 'crypto.pbkdf2 without specifying' +
-      ' a digest is deprecated. Please specify a digest', 'DEP0009');
-
-
 exports.pbkdf2 = function(password,
                           salt,
                           iterations,
                           keylen,
                           digest,
                           callback) {
   if (typeof digest === 'function') {
-    callback = digest;
-    digest = undefined;
-    pbkdf2DeprecationWarning();
+    // Formerly Deprecation code DEP0009
+    throw new TypeError(
+      'The "digest" argument is required and must not be undefined');
```

Comment:
It checks if `digest` is a function and throws error saying `must not be undefined`. I think this could do stricter validations.